test/test-apimatic-core/utilities/api_helper_test.rb

```diff
@@ -339,27 +339,6 @@ def test_complex_type_query_params
                   "query_param[name]"=>"Jone", "query_param[uid]"=>"1234", "query_param[personType]"=>"Per"})
 
   end
-  def test_deserialze
-    obj = ApiHelper.deserialize('anyOf(Integer, Atom)',
-                                                                '{ "NumberOfElectrons": 4,"NumberOfProtons": 3}',
-                                                                TestComponent, false)
-    assert obj.instance_of?(TestComponent::Atom)
-    assert_equal(ApiHelper.deserialize('anyOf(Float, String)','0.987', TestComponent, false),
-                 0.987)
-    assert_equal(ApiHelper.deserialize('anyOf(Float, String)',
-                          '"some string"', TestComponent, false),"some string")
-    obj = ApiHelper.deserialize('anyOf(Integer, Atom)',
-                          '{"NumberOfElectrons":4,"NumberOfProtons":3}', TestComponent, false
-    )
-    assert obj.instance_of?(TestComponent::Atom)
-    obj = ApiHelper.deserialize('oneOf(Morning, Atom, Car)',
-                          '{"startsAt":"6:00","endsAt":"11:00","offerTeaBreak":true,"sessionType":'\
-      '"Morning"}', TestComponent, false
-    )
-    assert obj.instance_of?(TestComponent::Morning)
-
-  end
-
   def test_json_serialize
```

Comment:
Blocker: The unit tests for the newly added methods like `valid_type`, `apply_primitive_converter`, `deserialize_union_type` are missing.

Reply:
done